modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java

```diff
@@ -78,278 +77,76 @@ public void onApplicationEvent(ApplicationEvent event) {
-     */
-    public void registerManagedService(String oclLocation) throws Exception {
-        Ocl ocl = this.oclLoader.getOcl(new URL(oclLocation));
-        registerManagedService(ocl);
-    }
-
-    /**
-     * Update existing managed service with a new/updated OCL descriptor, at the given location.
+     * Persist the result of the deployment.
      *
-     * @param managedServiceName the managed service to update, identified by the given name.
-     * @param oclLocation        the new/updated OCL descriptor location.
-     * @throws Exception if the update fails.
+     * @param deployResult the result of the deployment.
      */
-    public void updateManagedService(String managedServiceName, String oclLocation)
-            throws Exception {
-        Ocl ocl = oclLoader.getOcl(new URL(oclLocation));
-        updateManagedService(managedServiceName, ocl);
-    }
-
-    /**
-     * Update existing managed service with a new/updated OCL descriptor.
-     *
-     * @param managedServiceName the managed service to update, identified by the given name.
-     * @param ocl                the new/update OCL descriptor.
-     */
-    public void updateManagedService(String managedServiceName, Ocl ocl) {
-        MDC.put(TASK_TYPE, TaskType.UPDATE.toValue());
-
-        if (!managedServiceName.equals(ocl.getName())) {
-            log.warn("Can not update managed service {} with {}", managedServiceName,
-                    ocl.getName());
-        }
-
-        if (!this.databaseOrchestratorStorage.isExists(managedServiceName)) {
-            throw new EntityNotFoundException(
-                    "Managed service " + managedServiceName + " not found");
-        }
-
-        for (OrchestratorPlugin orchestratorPlugin : plugins) {
-            ServiceStatusEntity serviceStatusEntity =
-                    this.databaseOrchestratorStorage.getServiceDetailsByNameAndPlugin(
-                            managedServiceName, orchestratorPlugin);
-            putServiceInfoIntoLogMdc(serviceStatusEntity);
-            try {
-                serviceStatusEntity.setServiceState(ServiceState.UPDATING);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-                orchestratorPlugin.updateManagedService(managedServiceName, ocl);
-                serviceStatusEntity.setServiceState(ServiceState.UPDATED);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-            } catch (RuntimeException exception) {
-                updateFailedServiceStatusEntity(serviceStatusEntity, exception);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-                throw exception;
-            }
-        }
+    public void storeDeployService(DeployResult deployResult) {
+        DeployServiceEntity deployServiceEntity = new DeployServiceEntity();
+        deployServiceEntity.setVersion(deployResult.getTask().getCreateRequest().getVersion());
```

Comment:
same here. Should we write the DB entities before execution and later only update the state based on the result?